pulser/_seq_drawer.py

```diff
@@ -199,6 +201,31 @@ def phase_str(phi):
         a.set_ylabel(r'$\Omega$ (rad/µs)', fontsize=14, labelpad=10)
         b.set_ylabel(r'$\delta$ (rad/µs)', fontsize=14)
 
+        if draw_phase_area:
+            for pulse_num, seq_ in enumerate(seq._schedule[ch]):
+                # Select only `Pulse` objects
+                if isinstance(seq_.type, Pulse):
+                    phase_val = seq_.type.phase / np.pi
+                    area_val = seq_.type.amplitude.integral / np.pi
+                    x_plot = (seq_.ti + seq_.tf) / 2
+                    if (
+                        seq._schedule[ch][pulse_num-1].type == "target"
+                        or pulse_num % 2 == 0
+                    ):
+                        y_plot = np.max(seq_.type.amplitude.samples) / 2
+                    else:
+                        y_plot = np.max(seq_.type.amplitude.samples)
```

Comment:
This will work most of the time, but what if you have two very close pulses after a `target`, and the first one has `pulse_num` odd? This will make them both place the box in the middle, which will make it overlap, right?

A solution to this is to have a variable storing whether the last box was placed on the top or the center, and to use its value to decide where the current box should go. Of course, after a `target` you would overwrite it.

Reply:
I think it is addressed now..Thanks